### DISCUSSION TITLE
how to keep the backend authenticated?

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#3148] [Q&A]
I'm facing an annoying problem.
I modified the validity of access tokens (both admin and common user), now they are valid for 24 hours.

I have a backend that authenticates to the DB with an admin account, however, I am often facing "no authentication" errors.
then if I restart the backend, it solves.

I'm using this code to access the db in the backend:

```js
import PocketBase from 'pocketbase'
const host = process.env.POCKETBASE_URL
const pb = new PocketBase(host);
await pb.admins.authWithPassword(process.env.POCKETBASE_ADMIN_EMAIL, process.env.POCKETBASE_ADMIN_PASS);
pb.autoCancellation(false);

async function updateAuthStore() {
    let newData = await pb.admins.authRefresh()
    pb.authStore.save(newData)
}

setInterval(updateAuthStore, 4 * 60 * 60 * 1000);

export default pb
```

I even created a function that runs every 4 hours to update and save the authStore

however the problem persists.

am I missing something?

**Comment by ganigeorgiev** [comment]
How do you use the above `pb` instance?

Note that if you are using it also for regular user authentication, then it would be incorrect and unsafe because all users will write over the same `pb.authStore`. It is OK to have a global server-side JS SDK instance **only** if used as admin and there is no risk user requests to modify the `pb.authStore` state.

For admin-only client you can also check https://github.com/pocketbase/pocketbase/discussions/2922#discussioncomment-6470104.
The idea is that you have to call the `authMiddleware()` function before every node request.
It will take care to either authenticate with email/pass or refresh the token (_only when needed_), so you don't have to keep a timer.
